crates/astria-sequencer/src/service/consensus.rs

```diff
@@ -150,6 +145,22 @@ impl Consensus {
         })
     }
 
+    #[instrument(skip(self))]
+    async fn handle_prepare_proposal(
+        &mut self,
+        prepare_proposal: request::PrepareProposal,
+    ) -> response::PrepareProposal {
+        self.app.prepare_proposal(prepare_proposal).await
```

Comment:
My first reaction to this was negative, but looking at the code structure it looks like `App` is actually a part of the `Consensus` service and only used by it. Should we make `app.rs` a submodule of it? Why even have the `App` type at all? It feels like it's not necessary unless I am missing something?

Reply:
the other consensus service functions all call into `app` the same way. and yes it's only used by the consensus service, however I think we should definitely keep `app` separate as it cleanly separates out the application execution logic from all other logic (eg. the consensus service, which is more closely tied to the ABCI server). the `app` also handles the state delta (the successive state changes which occur during execution)

Reply:
Yeah, so the reason I wrote the comment was that I thought `App` is a concept/object that operates at a higher hierarchy than `Consensus`.

So it felt wrong that it's passed to `Consensus` which then gets to manipulate it. It felt like an inversion of responsibility (higher level delegating to lower level).

But it turns out that it's the other way round: `App` here is a part of `Consensus`, so the latter delegates to the former.

I'm fine leaving it as its own type, but IMO it should be defined under the consensus module.